File: session6/lecture152/beakjoon_1644.cpp
#include <bits/stdc++.h>
using namespace std;
bool che[4000001];
int n, a[2000001], p, lo, hi, ret, sum;

int main()
{
  scanf("%d", &n);

  // 에라토스테네스체 를 만드는 과정
  for (int i = 2; i <= n; i++)
  {
    if (che[i])
      continue;
    for (int j = 2 * i; j <= n; j += i)
    {
      che[j] = 1;
    }
  }
  for (int i = 2; i <= n; i++)
  {
    if (!che[i])
      a[p++] = i;
  }

  // 투포인터 과정
  while (1)
  {
    if (sum >= n)
      sum -= a[lo++];

    //  high 와 p(소수의 개수) 가 같다면, 루프를 빠져나온다.
    else if (hi == p)
      break;
    else
      sum += a[hi++];
    if (sum == n)
      ret++;
  }
  printf("%d\n", ret);
}

File: session6/lecture152/beakjoon_1644.js
const input = require('fs').readFileSync('/dev/stdin', 'utf8').trim().split('\n');
let line = 0;
const readLine = () => input[line++];

const n = Number(readLine());
const che = new Array(4000001).fill(false);
const a = [];
let lo = 0, hi = 0, ret = 0, sum = 0;

// 에라토스테네스체 를 만드는 과정
for (let i = 2; i <= n; i++) {
  if (che[i]) continue;
  for (let j = 2 * i; j <= n; j += i) {
    che[j] = true;
  }
}
for (let i = 2; i <= n; i++) {
  if (!che[i])
    a.push(i);
}

const p = a.length;

// 투포인터 과정
while (true) {
  if (sum >= n)
    sum -= a[lo++];
  //  high 와 p(소수의 개수) 가 같다면, 루프를 빠져나온다.
  else if (hi === p)
    break;
  else
    sum += a[hi++];
  if (sum === n)
    ret++;
}
console.log(ret);
